tests/perf/common/test_harness.cc

```diff
@@ -0,0 +1,393 @@
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact darma@sandia.gov
+//
+// *****************************************************************************
+//@HEADER
+*/
+
+#include "test_harness.h"
+
+#include <vt/collective/collective_ops.h>
+#include <vt/scheduler/scheduler.h>
+#include <vt/phase/phase_manager.h>
+#include <vt/utils/memory/memory_usage.h>
+
+#include <numeric>
+#include <fstream>
+
+namespace vt { namespace tests { namespace perf { namespace common {
+
+/////////////////////////////////////////////////
+///////////////      HELPERS      ///////////////
+/////////////////////////////////////////////////
+
+/**
+ * \brief Prints memory usage in human readable format
+ */
+static std::string GetFormattedMemUsage(std::size_t memory) {
+  auto const& best_mem = util::memory::getBestMemoryUnit(memory);
+
+  return fmt::format(
+    "{}",
+    debug::emph(
+      fmt::format("{:.6g}{}", std::get<1>(best_mem), std::get<0>(best_mem))));
+}
+
+/**
+ * \brief Helper function that converts the test results from all runs into a final vector,
+ * which contains mean/min/max values for each test result.
+ *
+ * \param[in] input_vec vector that contains test data (memory usage or timers) from all test runs
+ * By default all tests are run 50 times (see --vt_perf_num_runs flag) so this vector will have 50 elems,
+ * and each element is a vector of test results for that run.
+ *
+ * \param[in] populate function that will be called (for each test run) in order to generate the output vector.
+ * Takes three params:
+ *  1. Final test result
+ *  2. Test result for given test run
+ *  3. Number of test runs
+ *
+ * \return Vector with final test results (mean/min/max)
+ */
+template <typename InputT, typename OutputT>
+std::vector<OutputT> ProcessInput(
+  std::vector<std::vector<InputT>> const& input_vec,
+  std::function<void(OutputT&, InputT const&, std::size_t const)> populate) {
```

Comment:
Another style suggestion:
```suggestion
  std::function<void(OutputT&, InputT const&, std::size_t const)> populate
) {
```